### PR TITLE
[incubator/fluentd-cloudwatch] update apiVersion to be compatible with k8s >= 1.16

### PR DESCRIPTION
**Pull request by marcatos**
Hi @jmcarp here's a quick update to this chart

#### What this PR does / why we need it:
I updated the daemonset to be compatible with k8s version >= 1.16

#### Which issue this PR fixes
none

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by sc250024** [comment]
Yes please!

**Comment by Tanbouz** [comment]
I ran into `missing required field "selector" in io.k8s.api.apps.v1.DaemonSetSpec` with `apps/v1`

Might need something like this:
```
spec:
  selector:
    matchLabels:
      app: {{ template "fluentd-cloudwatch.name" . }}
```

Around [here](https://github.com/helm/charts/blob/d2e40c29d12fe55a6b48539126b24af3c6b8ede6/incubator/fluentd-cloudwatch/templates/daemonset.yaml#L10)

**Comment by marcatos** [comment]
> I ran into `missing required field "selector" in io.k8s.api.apps.v1.DaemonSetSpec` with `apps/v1`
> 
> Might need something like this:
> 
> ```
> spec:
>   selector:
>     matchLabels:
>       app: {{ template "fluentd-cloudwatch.name" . }}
> ```
> 
> Around [here](https://github.com/helm/charts/blob/d2e40c29d12fe55a6b48539126b24af3c6b8ede6/incubator/fluentd-cloudwatch/templates/daemonset.yaml#L10)

Thanks @Tanbouz, I'll give it a try

**Comment by GregSharpe1** [comment]
Yes please! Just upgraded to EKS 1.16 :(

**Comment by GregSharpe1** [comment]
Hey @marcatos, we're going through this exact issue :) And having a look [here](https://github.com/Hevelop/charts/blob/d2e40c29d12fe55a6b48539126b24af3c6b8ede6/incubator/fluentd-cloudwatch/templates/psp.yaml#L2), there's still use of `extensions/v1beta1` which seems to be deprecated in this [blog](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) I've found with `PodSecurityPolicies`.

**Comment by marcatos** [comment]
Just committed your suggested updates @Tanbouz @GregSharpe1, let's see if is enough to pass tests.

**Comment by sionsmith** [comment]
Hey guys is there any update on this? Or any current work around?

**Comment by jtai-omniex** [comment]
Any update? Really need this change

**Comment by marcatos** [comment]
@davidkarlsen @prydonius can you give us a feedback please?

File: incubator/fluentd-cloudwatch/templates/daemonset.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
```
Comment (davidkarlsen) [inline incubator/fluentd-cloudwatch/templates/daemonset.yaml:2]: maybe use capabilities instead to make it dynamic so you can target older and newer apis with same chart: https://helm.sh/docs/chart_template_guide/builtin_objects/ ?

File: incubator/fluentd-cloudwatch/templates/psp.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.rbac.pspEnabled }}
-apiVersion: extensions/v1beta1
```
Comment (davidkarlsen) [inline incubator/fluentd-cloudwatch/templates/psp.yaml:2]: maybe use capabilities instead to make it dynamic so you can target older and newer apis with same chart: https://helm.sh/docs/chart_template_guide/builtin_objects/ ?

**Review by davidkarlsen — CHANGES_REQUESTED** [review]
maybe use capabilities instead to make it dynamic so you can target older and newer apis with same chart: https://helm.sh/docs/chart_template_guide/builtin_objects/ ?

**Comment by marcatos** [comment]
Hi @davidkarlsen, 

I've just pushed a backward compatibility update as you suggested.

File: incubator/fluentd-cloudwatch/templates/daemonset.yaml [hunk 3]
```diff
@@ -1,4 +1,8 @@
-apiVersion: extensions/v1beta1
+{{- if .Capabilities.APIVersions.Has "apps/v1" }}
+apiVersion: apps/v1
+{{- else }}
+apiVersion: apps/v1beta1
```
Comment (davidkarlsen) [inline incubator/fluentd-cloudwatch/templates/daemonset.yaml]: shouldn't that be extensions/v1beta1?

**Comment by aschepp** [comment]
Hello, any update on this?

File: incubator/fluentd-cloudwatch/templates/daemonset.yaml (hunk 3 above)
Comment (sionsmith) [reply incubator/fluentd-cloudwatch/templates/daemonset.yaml]: Hi @marcatos is there anything I can help with on this?

Comment (marcatos) [reply incubator/fluentd-cloudwatch/templates/daemonset.yaml]: Hi @davidkarlsen @sionsmith, sorry for this huge delay, I've been busy this month. 
I've just pushed a commit to fix my copy/paste error.

**Pull request merged**